apps/src/levelbuilder/lesson-editor/ActivityCard.jsx

```diff
@@ -151,34 +152,44 @@ class ActivityCard extends Component {
         </div>
         <div style={styles.activityBody} hidden={this.props.collapsed}>
           {activity.activitySections.map(section => (
-            <ActivitySectionCard
-              key={section.key}
-              activitySection={section}
-              activityPosition={activity.position}
-              activitySectionsCount={activity.activitySections.length}
-              activitiesCount={this.props.activitiesCount}
-              ref={ref => {
-                setActivitySectionRef(ref, activity.position, section.position);
-              }}
-              activitySectionMetrics={this.props.activitySectionMetrics}
-              updateTargetActivitySection={updateTargetActivitySection}
-              clearTargetActivitySection={clearTargetActivitySection}
-              targetActivityPos={this.props.targetActivityPos}
-              targetActivitySectionPos={this.props.targetActivitySectionPos}
-              updateActivitySectionMetrics={updateActivitySectionMetrics}
-              hasLessonPlan={hasLessonPlan}
-              allowMajorCurriculumChanges={allowMajorCurriculumChanges}
-            />
+            <div key={section.key + '-container'}>
+              <ActivitySectionCard
+                key={section.key}
+                activitySection={section}
+                activityPosition={activity.position}
+                activitySectionsCount={activity.activitySections.length}
+                activitiesCount={this.props.activitiesCount}
+                ref={ref => {
+                  setActivitySectionRef(
+                    ref,
+                    activity.position,
+                    section.position
+                  );
+                }}
+                activitySectionMetrics={this.props.activitySectionMetrics}
+                updateTargetActivitySection={updateTargetActivitySection}
+                clearTargetActivitySection={clearTargetActivitySection}
+                targetActivityPos={this.props.targetActivityPos}
+                targetActivitySectionPos={this.props.targetActivitySectionPos}
+                updateActivitySectionMetrics={updateActivitySectionMetrics}
+                hasLessonPlan={hasLessonPlan}
+                allowMajorCurriculumChanges={allowMajorCurriculumChanges}
+              />
+              <div key={section.key + 'add'}>
```

Comment:
Maybe you don't need the extra `<div>` depth here and you could just put the `key={}` parameter on the `<button>` itself?

Alternatively and very optionally: this method has gotten out of control length-wise. Its long enough that I would consider splitting this out into a new functional component (in the same file) `InsertActivitySectionButton`:

```
const InsertActivitySectionButton = (handleAddActivitySection) => (
  <button>...</button>
);
```